src/Database/Schema/PostgresSchemaDialect.php

```diff
@@ -379,6 +384,12 @@ public function columnSql(TableSchema $schema, string $name): string
     {
         /** @var array $data */
         $data = $schema->getColumn($name);
+
+        $sql = $this->_getTypeSpecificColumnSql($data['type'], $schema, $name);
```

Comment:
Couldn't a custom type also need other attributes in the abstract schema? Keys like length, precision and null could be useful in a custom type.